hydra-plutus/src/Hydra/Contract/Commit.hs

```diff
@@ -107,20 +116,21 @@ validator (_party, _commit, headId) r ctx =
   headOutputValue =
     txOutValue . head $ txInfoOutputs (scriptContextTxInfo ctx)
 
-compiledValidator :: CompiledCode ValidatorType
+compiledValidator :: TypedValidator CommitValidator
```

Comment:
IIRC this was adding some overhead to the plutus scripts and that's why we ended up with "plain type aliases" on `DatumType` and `RedeemerType`.

Might not be the case anymore, but we should be cautious.